dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -0,0 +1,413 @@
+        log.warn("The relationshipType's right entityType label is: " + relationshipType.getRightType().getLabel());
+        log.warn("The relationshipType's left min cardinality is: " + relationshipType.getLeftMinCardinality());
+        log.warn("The relationshipType's left max cardinality is: " + relationshipType.getLeftMaxCardinality());
+        log.warn("The relationshipType's right min cardinality is: " + relationshipType.getRightMinCardinality());
+        log.warn("The relationshipType's right max cardinality is: " + relationshipType.getRightMaxCardinality());
+    }
+
+    private boolean verifyMaxCardinality(Context context, Item itemToProcess,
+                                         Integer maxCardinality,
+                                         RelationshipType relationshipType) throws SQLException {
+        List<Relationship> rightRelationships = findByItemAndRelationshipType(context, itemToProcess, relationshipType,
+                                                                              false);
+        if (maxCardinality != null && rightRelationships.size() >= maxCardinality) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean verifyEntityTypes(Item itemToProcess, EntityType entityTypeToProcess) {
+        List<MetadataValue> list = itemService.getMetadata(itemToProcess, "relationship", "type", null, Item.ANY);
+        if (list.isEmpty()) {
+            return false;
+        }
+        String leftEntityType = list.get(0).getValue();
+        if (!StringUtils.equals(leftEntityType, entityTypeToProcess.getLabel())) {
+            return false;
+        }
+        return true;
+    }
+
+    public Relationship find(Context context, int id) throws SQLException {
+        Relationship relationship = relationshipDAO.findByID(context, Relationship.class, id);
+        return relationship;
+    }
+
+    @Override
+    public List<Relationship> findByItem(Context context, Item item) throws SQLException {
+
+        List<Relationship> list = relationshipDAO.findByItem(context, item);
+
+        list.sort((o1, o2) -> {
+            int relationshipType = o1.getRelationshipType().getLeftLabel()
+                                     .compareTo(o2.getRelationshipType().getLeftLabel());
+            if (relationshipType != 0) {
+                return relationshipType;
+            } else {
+                if (o1.getLeftItem() == item) {
+                    return o1.getLeftPlace() - o2.getLeftPlace();
+                } else {
+                    return o1.getRightPlace() - o2.getRightPlace();
+                }
+            }
+        });
+        return list;
+    }
+
+    @Override
+    public List<Relationship> findAll(Context context) throws SQLException {
+        return relationshipDAO.findAll(context, Relationship.class);
+    }
+
+    public void update(Context context, Relationship relationship) throws SQLException, AuthorizeException {
```

Comment:
please, add the override annotation here and on the following methods

Reply:
This has been included in https://github.com/DSpace/DSpace/pull/2411